tests-with-coverage/Readme.md

```diff
@@ -0,0 +1,611 @@
+# Measuring the coverage of your tests
+
+Learn how to measure the test coverage of your application. This guide covers:
+
+* Measuring the coverage of your Unit Tests
+* Measuring the coverage of your Integration Tests
+* Separating the execution of your Unit Tests and Integration Tests
+* Consolidating the coverage for all your tests
+
+## Prerequisites
+
+To complete this guide, you need:
+
+* less than 15 minutes
+* an IDE
+* JDK 1.8+ installed with JAVA_HOME configured appropriately
+* Apache Maven 3.5.3+
+* Having completed the guide [Geeting started testing](https://quarkus.io/guides/getting-started-testing)
+
+## Architecture
+
+The application built in this guide is just a JAX-RS endpoint (hello world) that relies on dependency injection to use a service.
+Service will be tested thanks to JUnit 5 and Endpoint will be tested via a ```@QuarkusTest``` annotation.
+
+## Solution
+
+We recommend that you follow the instructions in the next sections and create the application step by step. However, you can go right to the completed example.
+Clone the Git repository: git clone {quickstarts-clone-url}, or download an {quickstarts-archive-url}[archive].
+The solution is located in the tests-with-coverage directory.
+
+## Starting from a simple project and two tests
+
+Let's start with an application that looks like the content of the guide *getting-started-testing* but with a few modifications for the sake of the example.
+First, an application serving a hello endpoint:
+
+```Java
+package org.acme.quickstart;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+@Path("/hello")
+public class GreetingResource {
+
+    private final GreetingService service;
+
+    @Inject
+    public GreetingResource(GreetingService service) {
```

Comment:
Can you explain to end users a bit why you don't do just

```java
    @Inject
    GreetingService service;
```

And maybe add a note that with the `@Inject GreetingService service;` approach test coverage is visible properly. Constructor based injection approach involves generated code which needs instrumentation.

Reply:
I intentionally skipped that part to avoid making things more complicated. But you're certainly right and the role of such an example is to explain things with as much details as possible.

I added some explanations at the end of the section *The coverage does not seem to correspond to the reality* . I hope it is clear enough. I did not remember *why* the no-argument constructor was injected (rest-easy?). I don't know if is necessary to mention it.

Reply:
A bit verbose in code examples, but it's not show stopper ;)